pandas/core/reshape/merge.py

```diff
@@ -1452,6 +1454,147 @@ def _validate(self, validate: str) -> None:
             raise ValueError("Not a valid argument for validate")
 
 
+def _chunks(lst, n):
+    for i in range(0, len(lst), n):
+        yield lst[i : i + n]
+
+
+_DEFAULT_LEFT_CHUNK_SIZE = _DEFAULT_RIGHT_CHUNK_SIZE = 1000
+
+
+# TODO: perform conditional merge as optimized cython,
+#       rather than chunked/filtered cross merge, also
+#       incorporate v2 conditional merge logic into _MergeOperation,
+#       rather than the subclass below
+class _ConditionalEnabledMergeOperation(_MergeOperation):
```

Comment:
Instead of subclassing, I think all this logic needs to live in `_MergeOperation`. The current pattern is that each `_MergeOperation` subclass has it's own external API function which won't be the case for this feature.

Reply:
Hi @mroeschke, did you get a chance to see the proposed justification for this approach here: https://github.com/pandas-dev/pandas/pull/42964#issuecomment-1001712098

Specifically this part which enumerates the perceived advantages to this approach:
"""
With this newly proposed way, (1) any future reimplementation of this feature will not need to untangle the current implementation from _MergeOperation (however simple the untangling may be), (2) all other subclasses of _MergeOperation will not inherit any of the custom merge logic that is currently irrelevant to them. (3) Any merging changes unrelated to custom merging can continue being made to _MergeOperation without having to worry about accounting for the custom merge logic (i.e. _MergeOperation can remain the defacto main merging operation class). And (4) pd.merge will remain a simple instantiation/method call of a _MergeOperation object.
"""

And thanks for all the feedback, I'll get to these items soon.

Reply:
@aa1371 the issue is that you are working on an internal aspect of pandas here, not building on top. We *need* to have minimal code that conforms with the existing structure. The code base is huge and deviations from this make things even more confusing.

So pls do as we ask here.

Reply:
I think https://github.com/pandas-dev/pandas/pull/42964#issuecomment-1001712098 is good _in theory_ (moreover refactoring `_MergeOperation` subclass hierarchy is probably needed), but I think it's clearer in the short term to follow the pattern of `_MergeOperation` despite the feature incompatibilities, etc.

Reply:
@jreback @mroeschke - understood, will incorporate into _MergeCondition instead. Thanks.

Reply:
updated. Tried to keep the conditional merge logic as isolated as possible, while still reusing some existing code